src/coreclr/jit/lower.cpp

```diff
@@ -6492,6 +6492,7 @@ bool Lowering::LowerUnsignedDivOrMod(GenTreeOp* divMod)
             divisorValue -= 1;
         }
 
+        divMod->ClearDivFlags();
```

Comment:
when we convert DIV into a different binops we should reset those, otherwise they might alias with other flags

Reply:
`ChangeOper` does this for you, so I think you should just replace the `SetOper` calls with `ChangeOper` instead.

Reply:
> `ChangeOper` does this for you, so I think you should just replace the `SetOper` calls with `ChangeOper` instead.

Good point! Addressed